### PR TITLE
Use own Pen (if set) in GeometryDrawing.GetBounds

### LINKED ISSUE #5249: Pen Thickness is not taking into account when calculating bounding boxes/clip rectangles.
When drawing e.g. an ellipse into an image with just the outline, it will always get cut off at the edges because the bounding box appears to only consider the "fill" area, not the whole outline.

A simple repro:
```xml
<Image Width="500" Height="500" Stretch="None">
	<Image.Source>
		<DrawingImage>
			<GeometryDrawing>
				<GeometryDrawing.Pen>
					<Pen Brush="#000000" Thickness="10"></Pen>
				</GeometryDrawing.Pen>
				<EllipseGeometry RadiusX="100" RadiusY="100"></EllipseGeometry>
			</GeometryDrawing>
		</DrawingImage>
	</Image.Source>
</Image>
```

This currently looks like this: 

If this is intentional behavior, what is the proper way to make this look like a normal circle?

### PR DESCRIPTION
**Pull request by Whiletru3**
## What does the pull request do?
Use the GeometryDrawing pen in GetBounds

## What is the current behavior?
The pen thickness wasn't use in GeometryDrawing.GetBounds, leading to a cropped image, see issue #5249 

## What is the updated/expected behavior with this PR?
Fix the GeometryDrawing

## How was the solution implemented (if it's not obvious)?
it is obvious :)
This file was renamed in master from Avalonia.Visuals to Avalonia.Base, the change here is only in 0.10.x but the same change should also be merged in the master if accepted.

## Checklist

- [x] Added unit tests (if possible)?

## Fixed issues
Fixes #5249

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/AvaloniaUI/Avalonia?pullRequest=8245) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/AvaloniaUI/Avalonia?pullRequest=8245) before we can accept your contribution.<br/><br/>:x: Whiletru3 [sign now](https://cla.dotnetfoundation.org/AvaloniaUI/Avalonia?pullRequest=8245)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/AvaloniaUI/Avalonia?pullRequest=8245) it.</sub>

**Comment by maxkatz6** [comment]
> the change here is only in 0.10.x but the same change should also be merged in the master if accepted.

Can you retarget this PR to the master branch? It should be merged there first.

File: src/Avalonia.Visuals/Media/GeometryDrawing.cs [hunk 1]
```diff
@@ -68,7 +68,8 @@ public override void Draw(DrawingContext context)
 
         public override Rect GetBounds()
         {
-            return Geometry?.GetRenderBounds(s_boundsPen) ?? Rect.Empty;
+            IPen pen = Pen ?? s_boundsPen;
```
Comment (maxkatz6) [inline src/Avalonia.Visuals/Media/GeometryDrawing.cs:71]: s_boundsPen has comment "// Adding the Pen's stroke thickness here could yield wrong results due to transforms." from 5 years ago. Sometime ago it was broken, and we discussed it's safer to merge it in master and see if there won't be a problem for users. Without backporting to a stable for now.

Comment (Whiletru3) [reply src/Avalonia.Visuals/Media/GeometryDrawing.cs:71]: @maxkatz6 , for now it is unusable in 0.10.14 like in this screenshot :
image

I can add this change in the master but the file moved to Avalonia.Base, do I need to create a second PR ? I'm not familiar with this ^^

Comment (maxkatz6) [reply src/Avalonia.Visuals/Media/GeometryDrawing.cs:71]: Yeah, I think there is no better way with github, but create another PR.

Comment (Whiletru3) [reply src/Avalonia.Visuals/Media/GeometryDrawing.cs:71]: > Also makes sense to add some simple tests.

I've add some tests (in Avalonia.RenderTests) :)

**Comment by maxkatz6** [comment]
@Whiletru3 @danwalmsley works as expected with render transforms and direct2d renderer, merging.

**Review by maxkatz6 — APPROVED** [review]

**Pull request merged**